plugin-admin/generate_plugins.js

```diff
@@ -0,0 +1,101 @@
+// Create empty object structure
+// Read plugins.ejs object in here and save as local variable
+// For each dep in package.json (as we don't want the ones not on npm):
+//   NOTE: scoped packages under node_modules/@seneca/title, unscoped under node_modules/name
+//   Write relevant information from package.json (stored in node_modules folder) to object
+//   Write relevant information not found in package.json from plugins.ejs variable to object
+//   NOTE: how to get groups if not every plugin.ejs is going in?
+// End for
+// Convert data to string to send
+// Write string data to file in /plugins-2023 to be included in another file in that dir
+const fs = require('fs')
+const jsonic = require('jsonic')
+
+let pluginData = {}
+
+let pkgjson = require('./package.json')
+Object.keys(pkgjson.dependencies).forEach((dep) => {
+  // npm fields: name, title, org_repo, desc
+  let pkg = require('./node_modules/' + dep + '/package.json')
+  // pkg.name
+  let title = pkg.name
+  switch (pkg.name[0]) {
+    case '@':
+      title = title.slice(8)
+      break
+    case 's':
+      title = title.slice(7)
+      break
+    default:
+      break
+  }
+  // org_repo
+  let org_repo = ''
+  if (typeof pkg.repository != 'undefined') {
+    let giturl = pkg.repository.url.split('/')
+    switch (giturl.length) {
+      case 5:
+        org_repo = giturl[3] + '/' + giturl[4].split('.')[0]
+        break
+      case 2:
+        giturl = pkg.repository.url.split('.')
+        org_repo = giturl[1].split(':')[1]
+        break
+      default:
+        break
+    }
+  }
+  pluginData[pkg.name] = {
+    title: title,
+    org_repo: org_repo,
+    desc: pkg.description,
+  }
+})
+
+// ejs fields: badges, badges urls
+let ejs = fs
+  .readFileSync('../src/pages/plugins-2023/plugins_man.ejs')
```

Comment:
plugins_man is not a great name - does not say very much about what this file is or contains - is it also a generate file?

Reply:
manual is not a great name - it exposes implementation details of the build process that are not relevant to the purpose of the file
perhaps: plugin_def ? these are the plugin definitions

Reply:
Sure, it could change in the future as only half the data is relevant (with the other half taken from npm for the generated include file)